core/services/feeds/service.go

```diff
@@ -596,9 +612,21 @@ func (s *service) ProposeJob(ctx context.Context, args *ProposeJobArgs) (int64,
 	if err != nil {
 		return 0, err
 	}
-
-	// Track the given job proposal request
-	promJobProposalRequest.Inc()
+	// auto approve workflow specs
+	if isWFSpec(args.Spec) {
+		promWorkflowRequests.Inc()
+		err = s.ApproveSpec(ctx, id, true)
```

Comment:
Is there any risk of a deadlock or other problems when you call ApproveSpec right away? I suspect that ProposeJob() wasn't expecting it to be called but maybe it's fine ...?

Reply:
i don't think so b/c this is outside the tx of the proposal. that is, proposal is complete or we don't get here.